EIPS/eip-4361-1.md

```diff
@@ -0,0 +1,238 @@
+
+The following is a non-normative example of a SIWE message with the SIWE CapGrok Extension:
+```
+example.com wants you to sign in with your Ethereum account:
+0x0000000000000000000000000000000000000000
+
+I further authorize https://example.com to perform the following actions on my behalf: (1) example: read for any. (2) example: append, delete for my.resource.1. (2) append for my.resource.2, my.resource.3.
+
+URI: https://example.com
+Version: 1
+Chain ID: 1
+Nonce: n-0S6_WzA2Mj
+Issued At: 2022-06-21T12:00:00.000Z
+Resources:
+- urn:capability:example:eyJkZWZhdWx0QWN0aW9ucyI6WyJyZWFkIl0sInRhcmdldGVkQWN0aW9ucyI6eyJteS5yZXNvdXJjZS4xIjpbImFwcGVuZCIsImRlbGV0ZSJdLCJteS5yZXNvdXJjZS4yIjpbImFwcGVuZCJdLCJteS5yZXNvdXJjZS4zIjpbImFwcGVuZCJdfX0
+```
+
+### CapGrok Capability
+
+A CapGrok Capibility is identified by their CapGrok URI that resolves to a CapGrok Details Object which defines the associated actions and optional target resources. The scope of each CapGrok Capability is attenuated by common fields in the SWIE message as described in the previous chapter, e.g., `URI`, `Issued At`, `Expiration Time`, `Not Before`.
+
+#### CapGrok URI Scheme
+
+A CapGrok URI starts with `urn:capability:` followed by the namespace discriminator, followed by `:` and the base64url-encoded payload of the CapGrok Details Object.
+
+The following is a non-normative example of a CapGrok Capability that uses the `example` namespace:
+```
+urn:capability:example:eyJkZWZhdWx0QWN0aW9ucyI6WyJyZWFkIl0sInRhcmdldGVkQWN0aW9ucyI6eyJteS5yZXNvdXJjZS4xIjpbImFwcGVuZCIsImRlbGV0ZSJdLCJteS5yZXNvdXJjZS4yIjpbImFwcGVuZCJdLCJteS5yZXNvdXJjZS4zIjpbImFwcGVuZCJdfX0
+```
+
+It is expected that RS implementers define their own namespace, e.g., `urn:capability:service:`. It is RECOMMENDED that RS implementers use the registry table in this specification to register their namespace and associated i8n translation file to avoid name collisions and/or name squatting.
+
+#### CapGrok Details Object Schema
+
+The CapGrok Details Object denotes which actions on which resources the delegee is authorized to invoke on behalf of the delegee for the validity period defined in the SIWE message. A CapGrok Details Object MUST follow the following JSON Schema:
+
+```
+{
+   "$schema":"http://json-schema.org/draft-04/schema#",
+   "type":"object",
+   "properties":{
+      "defaultActions":{
+         "type":"array",
+         "items":{
+            "type":"string"
+         },
+         "minItems":1
+      },
+      "targetedActions":{
+         "type":"object",
+         "patternProperties":{
+            "^.*$":{
+               "type":"array",
+               "items":{
+                  "type":"string"
+               },
+               "minItems":1
+            }
+         },
+         "minProperties":1
+      }
+   },
+   "minProperties":1,
+   "additionalProperties": false
+}
```

Comment:
Add `extraFields` to the JSON schema:
```suggestion
{
  "$schema": "http://json-schema.org/draft-04/schema#",
  "type": "object",
  "properties": {
    "defaultActions": {
      "type": "array",
      "items": {
        "type": "string",
        "minLength": 1
      },  
      "minItems": 1
    },  
    "targetedActions": {
      "type": "object",
      "patternProperties": {
        "^.*$": {
          "type": "array",
          "items": {
            "type": "string",
            "minLength": 1
          },  
          "minItems": 1
        }   
      },  
      "minProperties": 1
    },  
    "extraFields": {
      "type": "object",
      "minProperties": 1
    }   
  },  
  "minProperties": 1,
  "additionalProperties": false,
  "dependentSchemas": {
    "extraFields": {
      "minProperties": 2
    }   
  }
}
```

Reply:
Also improves validation for existing definitions:
- `defaultActions` actions must have a minimum length of 1.
- `targetedActions` targets must have a minimum length of 1.
- `targetedActions` actions must have a minimum length of 1.

Reply:
Should we also restrict the character set for actions and targets?